tests/util.getUserBestGearFromBank.test.ts

```diff
@@ -0,0 +1,253 @@
+				cape: { item: itemID('Cape of legends'), quantity: 1 },
+				hands: { item: itemID('Leather gloves'), quantity: 1 },
+				neck: { item: itemID('Occult necklace'), quantity: 1 },
+				head: { item: itemID('Helm of neitiznot'), quantity: 1 },
+				legs: { item: itemID('Ancestral robe bottom'), quantity: 1 },
+				body: { item: itemID('Ancestral robe top'), quantity: 1 }
+			},
+			toRemoveFromGear: resolveNameBank({
+				'Dragon chainbody': 1,
+				'Amulet of strength': 1,
+				'Elder maul': 1
+			}),
+			toRemoveFromBank: resolveNameBank({
+				'Helm of neitiznot': 1,
+				'Ancestral robe top': 1,
+				'Ancestral robe bottom': 1,
+				'Occult necklace': 1
+			}),
+			userFinalBank: resolveNameBank({
+				'Bandos chestplate': 4,
+				'Bandos tassets': 1,
+				'Helm of neitiznot': 1,
+				'Justiciar faceguard': 1,
+				'Dragon scimitar': 1,
+				'3rd age amulet': 1,
+				'Dragon chainbody': 1,
+				'Amulet of strength': 1,
+				'Elder maul': 1,
+				'Dragonfire shield': 1,
+				'Amulet of glory': 1
+			})
+		});
+	});
+	test('autoequip mage attack magic', async () => {
+		expect(
+			getUserBestGearFromBank(userBank, userGear, GearSetupTypes.Mage, 'attack', 'magic')
+		).toStrictEqual({
+			gearToEquip: {
+				...nullGear,
+				cape: { item: itemID('Cape of legends'), quantity: 1 },
+				hands: { item: itemID('Leather gloves'), quantity: 1 },
+				neck: { item: itemID('3rd age amulet'), quantity: 1 },
+				head: { item: itemID('Helm of neitiznot'), quantity: 1 },
+				legs: { item: itemID('Ancestral robe bottom'), quantity: 1 },
+				body: { item: itemID('Ancestral robe top'), quantity: 1 }
+			},
+			toRemoveFromGear: resolveNameBank({
+				'Dragon chainbody': 1,
+				'Amulet of strength': 1,
+				'Elder maul': 1
+			}),
+			toRemoveFromBank: resolveNameBank({
+				'Helm of neitiznot': 1,
+				'Ancestral robe top': 1,
+				'Ancestral robe bottom': 1,
+				'3rd age amulet': 1
+			}),
+			userFinalBank: resolveNameBank({
+				'Bandos chestplate': 4,
+				'Bandos tassets': 1,
+				'Helm of neitiznot': 1,
+				'Justiciar faceguard': 1,
+				'Dragon scimitar': 1,
+				'Occult necklace': 1,
+				'Dragon chainbody': 1,
+				'Amulet of strength': 1,
+				'Elder maul': 1,
+				'Dragonfire shield': 1,
+				'Amulet of glory': 1
+			})
+		});
+	});
+	test('autoequip melee defence slash', async () => {
+		expect(
+			getUserBestGearFromBank(userBank, userGear, GearSetupTypes.Melee, 'defence', 'slash')
+		).toStrictEqual({
+			gearToEquip: {
+				...nullGear,
+				body: { item: itemID('Bandos chestplate'), quantity: 1 },
+				cape: { item: itemID('Cape of legends'), quantity: 1 },
+				hands: { item: itemID('Leather gloves'), quantity: 1 },
+				neck: { item: itemID('Amulet of glory'), quantity: 1 },
+				head: { item: itemID('Justiciar faceguard'), quantity: 1 },
+				legs: { item: itemID('Bandos tassets'), quantity: 1 },
+				weapon: { item: itemID('Dragon scimitar'), quantity: 1 },
+				shield: { item: itemID('Dragonfire shield'), quantity: 1 }
```

Comment:
you can PR this later, but this would be a lot simpler with the `constructGearSetup` utility I made.